### PR TITLE
Fix runtime check during restore

### PR DESCRIPTION
**Pull request by ZeyadYasser**
[cfg.RuntimePath](https://github.com/containers/podman/blob/8e88abda85f7bf44b6857ad5d62c8ef58206fce5/cmd/podman/root.go#L146) was set to default runtime, so the empty string check fails. Instead we could check if the flag was changed.

It is a minor bug that was already mentioned in the issue below, so I didn't think I should create a new issue. Please let me know if this is a requirement.

Fixes: #11945

Signed-off-by: Zeyad Yasser <zeyady98@gmail.com>

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by ZeyadYasser** [comment]
[Runtime verification test for container checkpoint with export](https://github.com/containers/podman/blob/4274906a80c830492e0634602350c1626288e968/test/e2e/checkpoint_test.go#L1492)
used the default runtime for test which causes test to always
pass. Problem rises when using non-default runtime, then doing
a restore.
I just forced using a non-default runtime during container creation to trigger the edge case I faced.

**Comment by mheon** [comment]
@edsantiago PTAL

**Comment by ZeyadYasser** [comment]
https://api.cirrus-ci.com/v1/artifact/task/4817234857033728/ginkgo_node_logs/test/e2e/ginkgo-node-3.log

Looks like `--runtime` is implicitly set to the default runtime, I will look into a way to remove the `--runtime` arg.

**Comment by ZeyadYasser** [comment]
> https://api.cirrus-ci.com/v1/artifact/task/4817234857033728/ginkgo_node_logs/test/e2e/ginkgo-node-3.log
> 
> Looks like `--runtime` is implicitly set to the default runtime, I will look into a way to remove the `--runtime` arg.

UPDATE: I created a new test to address this problem.

File: test/e2e/checkpoint_test.go
Comment (edsantiago) [inline test/e2e/checkpoint_test.go:1640]: I know this is copy-paste from elsewhere, but you might take the opportunity to make this variable name more descriptive

File: test/e2e/checkpoint_test.go
Comment (edsantiago) [inline test/e2e/checkpoint_test.go:1610]: Okay, why `Contains`? You're asking for `.Name`. It should be safe to use `Equals` here.

File: test/e2e/checkpoint_test.go
Comment (edsantiago) [inline test/e2e/checkpoint_test.go:1611]: Hint: a future maintainer seeing this message would reeeeeeeeally appreciate something like "Default runtime is %q; this test requires crun to be default".

**Review by edsantiago — CHANGES_REQUESTED** [review]
Test LGTM overall, but needs some work. Thanks @ZeyadYasser 

(and OBTW I suspect the team will ask you to squash your commits)

**Review by edsantiago — APPROVED** [review]

**Comment by rhatdan** [comment]
LGTM but reverse your PRs.

**Comment by edsantiago** [comment]
@mheon here we go again with [the timeout](https://api.cirrus-ci.com/v1/task/6116050067521536/logs/main.log). I've restarted anyway.

**Comment by mheon** [comment]
@vrothberg Looks like we're seeing timeouts in `podman wait`, specifically in remote Podman, after the Gitlab patches landed.

**Comment by mheon** [comment]
At least seem to be significantly rarer than the other issues. I don't see any obvious cause, unfortunately.

**Pull request merged**
